hazelcast-jet-sql/src/test/java/com/hazelcast/jet/sql/impl/connector/kafka/SqlPrimitiveTest.java

```diff
@@ -326,38 +294,67 @@ public void test_explicitKeyAndThis() {
+        String name = createRandomTopic();
+        sqlService.execute("CREATE MAPPING " + name + ' '
+                + "TYPE " + KafkaSqlConnector.TYPE_NAME + ' '
+                + "OPTIONS ( "
+                + '"' + OPTION_KEY_FORMAT + "\" '" + JAVA_FORMAT + '\''
+                + ", \"" + OPTION_KEY_CLASS + "\" '" + Integer.class.getName() + '\''
+                + ", \"" + OPTION_VALUE_FORMAT + "\" '" + JAVA_FORMAT + '\''
+                + ", \"" + OPTION_VALUE_CLASS + "\" '" + String.class.getName() + '\''
+                + ", \"key.serializer\" '" + IntegerSerializer.class.getCanonicalName() + '\''
+                + ", \"key.deserializer\" '" + IntegerDeserializer.class.getCanonicalName() + '\''
+                + ", \"value.serializer\" '" + StringSerializer.class.getCanonicalName() + '\''
+                + ", \"value.deserializer\" '" + StringDeserializer.class.getCanonicalName() + '\''
+                + ", \"bootstrap.servers\" '" + kafkaTestSupport.getBrokerConnectionString() + '\''
+                + ", \"auto.offset.reset\" 'earliest'"
+                + ")"
+        );
+
+        assertTopicEventually(
+                name,
+                "INSERT INTO " + name + " (__key, this) VALUES (1, '2')",
+                createMap(1, "2")
+        );
+        assertRowsEventuallyInAnyOrder(
+                "SELECT * FROM " + name,
+                singletonList(new Row(1, "2"))
+        );
     }
 
     @Test
     public void test_noKeyFormat() {
         String topicName = createRandomTopic();
         sqlService.execute("CREATE MAPPING " + topicName + ' '
-                + "TYPE " + KafkaSqlConnector.TYPE_NAME + ' '
-                + "OPTIONS ("
-                + '"' + OPTION_VALUE_FORMAT + "\" '" + JAVA_FORMAT + "',"
-                + '"' + OPTION_VALUE_CLASS + "\" '" + Integer.class.getName() + "'"
-                + ", \"bootstrap.servers\" '" + kafkaTestSupport.getBrokerConnectionString() + '\''
-                + ", \"key.serializer\" '" + IntegerSerializer.class.getCanonicalName() + '\''
-                + ", \"key.deserializer\" '" + IntegerDeserializer.class.getCanonicalName() + '\''
-                + ", \"value.serializer\" '" + IntegerSerializer.class.getCanonicalName() + '\''
-                + ", \"value.deserializer\" '" + IntegerDeserializer.class.getCanonicalName() + '\''
-                + ", \"auto.offset.reset\" 'earliest'"
-                + ")");
+                           + "TYPE " + KafkaSqlConnector.TYPE_NAME + ' '
+                           + "OPTIONS ("
+                           + '"' + OPTION_VALUE_FORMAT + "\" '" + JAVA_FORMAT + "',"
+                           + '"' + OPTION_VALUE_CLASS + "\" '" + Integer.class.getName() + "'"
+                           + ", \"key.serializer\" '" + IntegerSerializer.class.getCanonicalName() + '\''
+                           + ", \"key.deserializer\" '" + IntegerDeserializer.class.getCanonicalName() + '\''
```

Comment:
Can't we remove these two lines?

Reply:
serializer options are required, without format we are unable to infer it